.github/workflows/release.yml

```diff
@@ -0,0 +1,40 @@
+name: Release
+on:
+  push:
+    tags:
+      - v*
+
+env:
+  ELIXIR_OPTS: "--warnings-as-errors"
+  ERLC_OPTS: "warnings_as_errors"
+  LANG: C.UTF-8
+
+jobs:
+  building_release:
+    strategy:
+      fail-fast: true
+      matrix:
+        otp_release: ["OTP-24.0"]
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 50
+      - name: Install Erlang/OTP
+        run: |
+          cd $RUNNER_TEMP
+          wget -O otp.tar.gz https://repo.hex.pm/builds/otp/ubuntu-18.04/${{ matrix.otp_release }}.tar.gz
+          mkdir -p otp
+          tar zxf otp.tar.gz -C otp --strip-components=1
+          otp/Install -minimal $(pwd)/otp
+          echo "$(pwd)/otp/bin" >> $GITHUB_PATH
+      - name: Build Elixir Release
+        run: |
+          make Precompiled.zip
+          mv Precompiled-v1.14.0-dev.zip elixir-${{ github.ref_name }}-${{ matrix.otp_release }}.zip
+          ls -la
+      - name: Upload Pre-built
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh release upload ${{ github.ref_name }} elixir-${{ github.ref_name }}-${{ matrix.otp_release }}.zip
```

Comment:
double-checking, do we need to first create a release off of a tag before we can upload assets to it?

Reply:
Yes. Currently, I assume that the maintainer should create a release first. After created, github will publish create tag event to the CI.

Reply:
Another question, do we need to rely on `VERSION` file instead of ref_name?

Reply:
i believe either is fine

Reply:
I'll let maintainers decide whether they would rather push a tag or a release. :)

Regardless how we get there, I think it would be preferred if the release starts as a _draft_ (e.g.: `$ gh release create --draft`) and it would only be _published_ (either manually or automatically) after all per-OTP builds have been uploaded. This way we'd only fire off release notifications when everything is ready.

Reply:
Thanks! I would try testing with this approach first.

Reply:
> Regardless how we get there, I think it would be preferred if the release starts as a draft (e.g.: $ gh release create --draft) and it would only be published (either manually or automatically) after all per-OTP builds have been uploaded.

Exactly! I think some manual steps may still be required, unfortunately, but this will reduce them greatly! ❤️

Reply:
> I'll let maintainers decide whether they would rather push a tag or a release. :)
> 
> Regardless how we get there, I think it would be preferred if the release starts as a _draft_ (e.g.: `$ gh release create --draft`) and it would only be _published_ (either manually or automatically) after all per-OTP builds have been uploaded. This way we'd only fire off release notifications when everything is ready.

As I do some research, the draft release will not create a tag, so it will not trigger a tag event and per [release event documentation](https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#release), it doesn't trigger if it's draft release. :(

Reply:
Good find. So yeah, I believe our trigger needs to be a tag push event from maintainers executing `$ git push origin <tag>` and similar. And then they'd either manually create a release draft or more likely we'd do it for them in the job. (so that by the time the job executes, there would be a release to attach files to.)

Reply:
Thanks for your idea! I try by create a job that create a draft  release first. After draft release is created, it'll upload the asset to the draft release. The draft release will looks like:

<img width="1153" alt="Screen Shot 2564-12-28 at 18 02 25" src="https://user-images.githubusercontent.com/484530/147559818-3c78b854-193b-4e43-ad81-3aff7dff5131.png">

Here a link to draft release https://github.com/wingyplus/elixir/releases/tag/untagged-358a0dfc87f7628dd7c8. I'm not sure anyone can see the draft or not. So I attach screen for demo above ☝️ 

Here is the pipeline https://github.com/wingyplus/elixir/actions/runs/1630211055: 

<img width="1303" alt="Screen Shot 2564-12-28 at 18 04 52" src="https://user-images.githubusercontent.com/484530/147559936-f15c43a4-79e3-4e22-bc3d-914da69e1a4a.png">

Reply:
I am not able to see the draft so thank you for posting the screenshot. Looks fantastic!